### ISSUE TITLE
Why my mintty cannot run windows commands

### ISSUE THREAD

**Issue by yang286** [mintty/mintty#658]
Hey everyone. Im Roy.
I got mintty shell after I installed Github. I didnot install CYGWIN.
If I run mintty from Github which i click 'open git shell', then I can run all windows commands.
If i run mintty by itself, then I cannot use any windows commands, why? It is so weird.
Here is the pic.
[http://blog.163.com/dsh808@126/prevPhDownload.do?host=dsh808@126&albumId=310061228&photoId=9790657658](url)
The left side is that i opened by mintty itself, so all windows command cannot use.
The right side is that i opened from Github, all windows commands no problem.
Thanks for any help.
by the way, why I cannot add my pic here??

**Comment by yang286** [comment]
help please

**Comment by mintty** [comment]
Please describe more clearly, or provide a screenshot :/

**Comment by yang286** [comment]
Hey Sir, thanks for your help. this windows I opened from gitshell, so all the commands are no problem.
![gitshell](https://cloud.githubusercontent.com/assets/28585814/25955280/60fd4f38-369b-11e7-9e6c-cb8954e4bf21.png)

And This window I opened from Mintty, but only little commands can be used. this is my problem. How can I use the commands such like dir, ls etc? thanks so much sir.
![mintty](https://cloud.githubusercontent.com/assets/28585814/25955447/e24085ba-369b-11e7-9192-4998a04076ed.png)

**Comment by yang286** [comment]
this 2 windows should be same things i guess, because they are from same window in the bottom you can see. This is very strange to me. 
![mintty](https://cloud.githubusercontent.com/assets/28585814/25957538/24fa74ec-36a1-11e7-9b87-991b37e6c8da.png)

**Comment by mintty** [comment]
If `ls` and most basic commands are not found, most likely /usr/bin is missing in your PATH variable.
However, this does not happen here if I start Git Bash, and start another mintty from there with Alt+F2 or a `mintty` command. My question remains unanswered, how *precisely* you started the disfunctional terminal; "opened from mintty" is quite fuzzy.

**Comment by yang286** [comment]
Hi Sir, this is where I started the Mintty, and I had made the /usr/bin in my system PATH variable.
C:\Users\roy\AppData\Local\GitHub\PortableGit_f02737a78695063deace08e96d5042710d3e32db\usr\bin      This fold had already been in my system PATH variable.
![mintty](https://cloud.githubusercontent.com/assets/28585814/25993258/050bb39e-373c-11e7-90f9-d8a6a7f52108.png)

**Comment by mintty** [comment]
That is not the intended way to start mintty and in fact /usr/bin is not included in PATH in that case.
You should start mintty with the "Cygwin Terminal" shortcut from the start menu or desktop.

Anyway, if you set PATH before starting mintty we should check why this is not effective. Please do this:
* start mintty as shown above, then type `echo $PATH`
* alternatively, start a Windows cmd window (Windows-key+R, cmd, Enter), then type `echo %PATH%`

**Comment by yang286** [comment]
Thank you Sir.  I just install a new GITHUB, then I get a MINGW64 terminal software after I install the new github.
In MINGW64 I dont have any problem to run all the command. So no everything is good. Thanks a lot Sir.
By the way, what is the difference between them? I found mintty and mingw64 have same logo. but mingw64 has no problem to run win cmd. Where you from Sir? Do you know where I can get a good forum to get help when I learn PERL PHP MYSQL etc program lauguages?

**Comment by mintty** [comment]
You mean Git for Windows, I suppose. It focusses on git and is based on MinGW. The difference is basically the package selection and the maintenance of git functionality.

**Comment by ying-zhang** [comment]
@yang286 

As the [above comment](https://github.com/mintty/mintty/issues/658#issuecomment-301140975) said, when you start `mintty.exe` directly (by double click the executable file or a shortcut), it loads a different `$PATH` environment variable comparing with starting by `git-shell`. 

You could check the `$PATH` environment variable by `echo $PATH` in `mintty`. 
You could update the default `$PATH` of `mintty` by editing `~/.bashrc` in the `mintty`.
**NOTE: NOT Windows's `%PATH%` environment variable** 

**You can execute any Windows commands under `mintty` if it could be found in mintty's `$PATH` (Again, NOT Windows's `%PATH%`)**

For example. If you installed `python for windows` at `C:/Python37`. You could execute `python` in Windows cmd because this path is added to Windows's `%PATH%` by the setup program. 
But you could not execute this python under `mintty`, if it's path is not appeared in `mintty`'s $PATH. 
You should update `mintty`'s $PATH by adding `export PATH=/c/Python37/bin:$PATH` to `~/.bashrc`, then reopen `mintty`.

@yang286 ，不知现在你的问题解决了没有。
正如[上面的评论](https://github.com/mintty/mintty/issues/658#issuecomment-301140975) 所说的，不同的启动方式，`mintty`使用的 `$PATH` 环境变量（**跟Windows的 `%PATH%`环境变量无关**）是不同的。
你可以在`mintty`中执行`echo $PATH`查看当前的值。
只要你将对应Windows程序的路径添加 **`mintty`的`$PATH`** ，这样`mintty`可以找到对应的程序，就执行它了。

举个例子，假如你安装了Windows版的Python到`C:/Python37`，因为安装程序已经更新了Windows的`%PATH%`环境变量，所以可以直接在cmd执行`python`。
但`mintty`的`$PATH`没有更新这个安装目录，所以无法执行`python.exe`。
需要手动在`mintty`更新`$PATH`，在`~/.bashrc`添加一行`export PATH=/c/Python37/bin:$PATH`（注意路径变化和大小写），然后重新打开`mintty`，就可以直接执行Windows版的`python.exe`了。

**Comment by mintty** [comment]
When starting mintty, Windows %PATH% is propagated into the cygwin (or msys/git bash) environment as $PATH. Path syntax is properly converted on this occasion.
It is then further propagated to the shell started inside mintty. The shell startup scripts typically add further path components, like /bin, especially when run as a login shell. In theory, they could also remove path components. This is out of scope of mintty.
Apart from this general explanation, I am not sure what your comment intends to suggest.
